tendrl/node_agent/manager/manager.py

```diff
@@ -227,6 +228,34 @@ def on_pull(self, raw_data):
                     node_id=raw_data["node_id"],
                 )
             )
+        if "disks" in raw_data:
+            try:
+                self.etcd_client.delete(
+                    ("nodes/%s/Disks") % raw_data["node_id"], recursive=True)
```

Comment:
Lets take example, node have two disk vdb and vdc. When node_agent start these two disk details are stored in etcd. if vdb removed, then vdc details only updated and vdb details are not affected in etcd. Because as per etcd key details goes to vdc only. No vdb details comes, so vdb never affected. To solve this problem when ever any change comes in local inventory file and node inventory list, I am just deleting disk details and storing fresh details in etcd. So only vdc disk details are in etcd.